nearcache/config.go

```diff
@@ -0,0 +1,256 @@
+	}
+	if _, err := check.NonNegativeInt32(c.TimeToLiveSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.Config.TimeToLiveSeconds: out of range", err)
+	}
+	if _, err := check.NonNegativeInt32(c.MaxIdleSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.Config.MaxIdleSeconds: out of range", err)
+	}
+	return nil
+}
+
+/*
+SetInvalidateOnChange sets if Near Cache entries are invalidated when the entries in the backing data structure are changed (updated or removed).
+When this setting is enabled, a Hazelcast instance with a Near Cache listens for cluster-wide changes on the entries of the backing data structure.
+And invalidates its corresponding Near Cache entries.
+Changes done on the local Hazelcast instance always invalidate the Near Cache immediately.
+Invalidate on change is true by default.
+*/
+func (c *Config) SetInvalidateOnChange(enabled bool) {
+	c.invalidateOnChange = &enabled
+}
+
+// InvalidateOnChange returns true invalide on change is enabled.
+// See the documentation for SetInvalidateOnChange.
+func (c Config) InvalidateOnChange() bool {
+	if c.invalidateOnChange == nil {
+		return true
+	}
+	return *c.invalidateOnChange
+}
+
+/*
+EvictionConfig is the configuration for eviction.
+
+You can set a limit for number of entries.
+The default values of the eviction configuration are:
+
+	* EvictionPolicyLRU as eviction policy
+	* MaxSizePolicyEntryCount as max size policy
+	* 2147483647 as maximum size for on-heap Map
+	* 10_000 as maximum size for all other data structures and configurations
+
+Eviction policy and comparator are mutually exclusive.
+*/
+type EvictionConfig struct {
+	evictionPolicy *EvictionPolicy
+	size           *int32
+	comparator     EvictionPolicyComparator
+	err            error
+}
+
+// Clone returns a copy of the configuration.
+func (c EvictionConfig) Clone() EvictionConfig {
+	return c
+}
+
+// Validate validates the configuration and sets the defaults.
+func (c *EvictionConfig) Validate() error {
+	if c.err != nil {
+		return c.err
+	}
+	if c.evictionPolicy != nil && c.comparator != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.EvictionConfig: only one of EvictionPolicy or Comparator can be configured", nil)
+	}
+	return nil
+}
+
+// SetEvictionPolicy sets the eviction policy of this eviction configuration.
+// The default policy is EvictionPolicyLRU which evicts the least recently used entries.
+func (c *EvictionConfig) SetEvictionPolicy(policy EvictionPolicy) {
+	if policy < 0 || policy >= evictionPolicyCount {
+		c.err = ihzerrors.NewInvalidConfigurationError("nearcache.EvictionConfig.SetEvictionPolicy: invalid policy", nil)
+		return
+	}
+	c.evictionPolicy = &policy
+}
+
+// EvictionPolicy returns the eviction policy of this eviction configuration.
+// See the documentation for SetEvictionPolicy.
+func (c EvictionConfig) EvictionPolicy() EvictionPolicy {
+	if c.evictionPolicy == nil {
+		return defaultEvictionPolicy
+	}
```

Comment:
Why don't we set defaults in Validate() for EvictionPolicy?. For example for `size` we say the default is 10000 but if the user checks the struct value it is `nil`. Isn't it confusing?

Reply:
`size` is not public to the user, so they cannot check its value.

Reply:
can't they debug and see its value?

Reply:
Yes, but if they can debug, they probably know about the getter too.